### DISCUSSION TITLE
Comprehensiv knowledge source

### DISCUSSION THREAD

**Discussion by stefannikolei** [SixLabors/ImageSharp#2389] [Ideas] [answered]
Perhaps it would be a good idea to start a collection of the multiple knowledge sources.

E.g. Reference Implementation, Specs etc.

It would probably also be good to have a list of a common tool collection.
 I often read, that someone tested it in tool x or y, but have never heard of that tool, or would not even had an idea to look for that ;)

For me this would be really useful while trying to contribute some time. 
But the initial knowledge which is needed to really contribute some new features is a real stopper in my eyes.

**Comment by antonfirsov** [comment]
You mean a proper developer guide right? That makes sense, though it would take non-trivial amount of time to cover everything, not sure if it's realistic.

Explanation of the terms you mentioned:
- Specs: this means open standard specifications, typically codec specification. Eg the [PNG spec](https://www.w3.org/TR/2003/REC-PNG-20031110/) or the [JPEG spec](https://www.w3.org/Graphics/JPEG/jfif3.pdf)
- Reference Implementation:
This could mean two things for me. (1) In general, it means an implementation of a Specification that is accepted by the community as "the correct one". Eg. [libpng ](http://www.libpng.org/pub/png/libpng.html) is the reference implementation of PNG if I'm not mistaken. (2) In ImageSharp context it could also mean any implementation of a feature that lives outside the ImageSharp codebase, eg. codec implementations from [ImageMagick.NET](https://github.com/dlemstra/Magick.NET) or unoptimized implementations of algorithms living in the test code eg. for [DCT](https://github.com/SixLabors/ImageSharp/blob/main/tests/ImageSharp.Tests/Formats/Jpg/Utils/ReferenceImplementations.AccurateDCT.cs). We typically use these in our tests to compare the outputs of our actual implementations to the outputs of the reference implementations.

Can you list further (specific!) topics, terms, tools, you are curious about?

**Reply by stefannikolei** [reply]
I would not call it a Developer guide. Probably just a good collection of content for this complex knowledge space :) 


E.g. there was a app called gifiddle or https://ezgif.com/ in #2198

**Comment by br3aker** [comment, marked as answer]
Thumbs up for a 'tool collection', I believe it can have its space somewhere in the wiki :)

Not so sure about other knowledge resources, while it's a great concept on paper it lacks motivation in my opinion. If somebody has a deep undestanding in feature X why would somebody spend time on documentation of how things work instead of actually developing said things?

I do understand the idea, I just can't imagine how this can be accomplished.

**Reply by stefannikolei** [reply]
Yeah I totally understand that. My intention was not to have someone write those things up. Rather create a link collection to useful resources which are known by you. Those can probably help contributors to have a starting point for learning/investigation.

**Reply by brianpopow** [reply]
@br3aker and @stefannikolei:  I like the idea of a wiki page for a collection of useful tools and links. I will start it this weekend.

**Reply by brianpopow** [reply]
here is a first version: https://github.com/SixLabors/ImageSharp/wiki/Useful-tools-and-links

Let me know, if I should add anything.

**Reply by JimBobSquarePants** [reply]
Is there any way we can get a compiled version of the tiff tools stashed somewhere? I can't see a link on the site only links to zipped releases of the sources.

**Reply by brianpopow** [reply]
On ubuntu you can get the tiff tools via:
```
apt-get install libtiff-tools
```

But I guess you are on windows. I dont think you can get the binary files directly. You will need to compile them yourself.
You will need Cmake for that.
Then in the libtiff source folder:
```
cmake-gui .
```

change build directory to libtiff/bin

Then hit configure button, followed by generate and the you should be able to build the project with visual studio.

**Comment by AdjunProfPlum** [comment]
> Is there any way we can get a compiled version of the tiff tools stashed somewhere? I can't see a link on the site only links to zipped releases of the sources.

I recently did a debug build (including tiffd.dll, tiffcp.exe, tiffdump.exe, tiffinfo.exe, tiffset.exe, tiffsplit.exe and corresponding pdb files) on Windows.  Are you still interested?
[libtiff-tools.zip](https://github.com/SixLabors/ImageSharp/files/14289587/libtiff-tools.zip)

**Reply by JimBobSquarePants** [reply]
AWESOME! 👏

**Reply by JimBobSquarePants** [reply]
I'm getting a virus warning downloading these.
